ledger/participant-integration-api/src/main/scala/platform/apiserver/services/ApiSubmissionService.scala

```diff
@@ -105,114 +104,125 @@ private[apiserver] final class ApiSubmissionService private (
     commandExecutor: CommandExecutor,
     configuration: ApiSubmissionService.Configuration,
     metrics: Metrics,
-)(implicit ec: ExecutionContext, mat: Materializer, logCtx: LoggingContext)
+)(implicit ec: ExecutionContext, mat: Materializer, loggingContext: LoggingContext)
     extends CommandSubmissionService
     with ErrorFactories
     with AutoCloseable {
 
   private val logger = ContextualizedLogger.get(this.getClass)
 
+  private val DuplicateCommand = Status.ALREADY_EXISTS.augmentDescription("Duplicate command")
+
+  override def submit(request: SubmitRequest): Future[Unit] =
+    withEnrichedLoggingContext(logging.commands(request.commands)) { implicit loggingContext =>
```

Comment:
This is much nicer. Thanks!

Reply:
I'll use this approach to make the other calls to `withEnrichedLoggingContext` a bit more compact.